tests/extmod/socket_fileno.py

```diff
@@ -0,0 +1,17 @@
+# Test socket.fileno() functionality
+
+try:
+    import socket, errno
+except ImportError:
+    print("SKIP")
+    raise SystemExit
+
+if not hasattr(socket.socket, "fileno"):
+    print("SKIP")
+    raise SystemExit
+
+s = socket.socket()
+assert s.fileno() >= 0
+
+s.close()
+assert s.fileno() == -1
```

Comment:
```suggestion
print(s.fileno())  #  should print -1
```

Could even do this one like this since it is an exact value.